docs/running-multi-parsers.rst

```diff
@@ -216,6 +218,16 @@ higher level language, a subparser for the higher level language is
 chosen.  Choosing a parser in this non-deterministic way (dynamic way)
 is called *bottom up*.
 
+	============================== ===================== ======================
+	input file                     tags of *<BASE_LANG>* tags of *<SUB_LANG>*
+	============================== ===================== ======================
+	*<BASE_LANG>* (``input.base``) top down              bottom up
+	*<SUB_LANG>*  (``input.sub``)  bottom up             top down
+	============================== ===================== ======================
+
+.. TODO: Q to @masatake san: Is this correct?
+	Especially "bottom up" in the middle of the bottom line is not sure.
+
```

Comment:
I will answer to this question. Please, wait for awhile.

Reply:
Please take your time.  Thank you.

Reply:
I will review this in this week end.

Reply:
Thank you.  But please respond at your convenience.

I've been busy with other works.
I have noticed after seeing my PR today there is a sentence in the first line of section "Top down parser choice and bottom up parser choice";

> There are two ways to run a subparser: top down or bottom up parser choices.

This means terms "top down" and "bottom up" are ones for a subparser.

Therefore the following table I proposed in this PR;

```
============================== ===================== ======================
input file                     tags of *<BASE_LANG>* tags of *<SUB_LANG>*
============================== ===================== ======================
*<BASE_LANG>* (``input.base``) top down              bottom up
*<SUB_LANG>*  (``input.sub``)  bottom up?            top down
============================== ===================== ======================
```

should be

```
============================== =================
input file                     subparser choices
============================== =================
*<SUB_LANG>*  (``input.sub``)  top down
*<BASE_LANG>* (``input.base``) bottom up
============================== =================
```

And the section "Top down parser choice and bottom up parser choice", which includes the table above, describes the implementation in C. Therefore, I reconsidered that it is better to move it to "API for subparser" in internal.rst.

Am I correct?

Reply:
I found the quality of implementation and *concept* of the "direction" is very low.
I must refine the concept itself.

```
============================== =================
input file                     subparser choices
============================== =================
*<SUB_LANG>*  (``input.sub``)  top down
*<BASE_LANG>* (``input.base``) bottom up
============================== =================
```

This looks correct.

> And the section "Top down parser choice and bottom up parser choice", which includes the table above, describes the implementation in C. Therefore, I reconsidered that it is better to move it to "API for subparser" in internal.rst.

Developers of optlib parsers should know this to use the direction flags. So I don't think the section should be move.

Reply:
> This looks correct.

Thanks.  I will update the commit.

> Developers of optlib parsers should know this to use the direction flags. So I don't think the section should be move.

Let me explain some more why I proposed moving the section.

- "Defining a subparser" section in optlib.rst is well-described without using words "top down" nor "bottom up". 
- I think a developer who needs subparser should use optlib, because is much more easy and simple.
- Only maintainers of C-API for subparser should know the concept of "top down" and "bottom up".

How do you think?

Reply:
I see. Please, move the section.

A developer of an optlib parser should know "top down" and "bottom up" choosing. I wanted to show you an example. However, it doesn't work well because of a bug.

Reply:
I've update the table as we agreed.

> I see. Please, move the section.
> 
> A developer of an optlib parser should know "top down" and "bottom up" choosing. I wanted to show you an example. However, it doesn't work well because of a bug.

I left the section as is, because it will be required when the bug is fixed.